node/src/Commands.ts

```diff
@@ -1961,6 +1973,124 @@ export function createGeoHash(
     return createCommand(RequestType.GeoHash, args);
 }
 
+// This type unites `GeoSearchResultOptions` and `GeoSearchOptions` of the java client
+/**
+ * Optional parameters for {@link BaseClient.geosearch|geosearch} command which defines what should be included in the
+ * search results and how results should be ordered and limited.
+ */
+export type GeoSearchResultOptions = {
+    /** Include the coordinate of the returned items. */
+    readonly withCoord?: boolean;
+    /**
+     * Include the distance of the returned items from the specified center point.
+     * The distance is returned in the same unit as specified for the `searchBy` argument.
+     */
+    readonly withDist?: boolean;
+    /** Include the geohash of the returned items. */
+    readonly withHash?: boolean;
+    /** Indicates the order the result should be sorted in. */
+    readonly sortOrder?: SortOrder;
+    /** Indicates the number of matches the result should be limited to. */
+    readonly count?: number;
+    /** Whether to allow returning as enough matches are found. This requires `count` parameter to be set. */
+    readonly isAny?: boolean;
+};
+
+/** Defines the sort order for nested results. */
+export enum SortOrder {
+    /** Sort by ascending order. */
+    ASC = "ASC",
+    /** Sort by descending order. */
+    DESC = "DESC",
+}
+
+export type GeoSearchShape = GeoCircleShape | GeoBoxShape;
+
+/** Circle search shape defined by the radius value and measurement unit. */
+export type GeoCircleShape = {
+    /** The radius to search by. */
+    readonly radius: number;
+    /** The measurement unit of the radius. */
+    readonly unit: GeoUnit;
+};
+
+/** Rectangle search shape defined by the width and height and measurement unit. */
+export type GeoBoxShape = {
+    /** The width of the rectangle to search by. */
+    readonly width: number;
+    /** The height of the rectangle to search by. */
+    readonly height: number;
+    /** The measurement unit of the width and height. */
+    readonly unit: GeoUnit;
+};
+
+export type SearchOrigin = CoordOrigin | MemberOrigin;
+
+/** The search origin represented by a {@link GeospatialData} position. */
+export type CoordOrigin = {
+    /** The pivot location to search from. */
+    readonly position: GeospatialData;
+};
+
+/** The search origin represented by an existing member. */
+export type MemberOrigin = {
+    /** Member (location) name stored in the sorted set to use as a search pivot. */
+    readonly member: string;
+};
+
+/**
+ * @internal
+ */
+export function createGeoSearch(
+    key: string,
+    searchFrom: SearchOrigin,
+    searchBy: GeoSearchShape,
+    resultOptions?: GeoSearchResultOptions,
+): command_request.Command {
+    let args: string[] = [key];
+
+    if ("position" in searchFrom) {
```

Comment:
Can we have a different check here? Checking for string seems incorrect here.

Reply:
It checks for the property named in the variable. `typeof`/`instanceof` doesn't work there unfortunately. Do you have a better idea how to do a runtime type check?

Reply:
this is the standard way to do it